test/recipes/25-test_x509.t

```diff
@@ -251,6 +263,36 @@ ok(run(app(["openssl", "x509", "-req", "-text", "-CAcreateserial",
             "-in", $b_csr])));
 ok(-e $ca_serial_dot_in_dir);
 
+# Tests for explict start and end dates of certificates
+ok(run(app(["openssl", "x509", "-req", "-text",
+	    "-key", $b_key,
+	    "-not_before", "20231031000000Z",
+	    "-not_after", "today",
+            "-in", $b_csr, "-out", $b_cert])));
+ok(get_not_before($b_cert) =~ /Oct 31 00:00:00 2023 GMT/);
+# explicit start and end dates
+ok(run(app(["openssl", "x509", "-req", "-text",
+	    "-key", $b_key,
+	    "-not_before", "20231031000000Z",
+	    "-not_after", "20231231000000Z",
+	    "-days", "99",
+            "-in", $b_csr, "-out", $b_cert])));
+ok(get_not_before($b_cert) =~ /Oct 31 00:00:00 2023 GMT/);
+ok(get_not_after($b_cert) =~ /Dec 31 00:00:00 2023 GMT/);
+# start date today and days
+ok(run(app(["openssl", "x509", "-req", "-text",
+	    "-key", $b_key,
+	    "-not_before", "today",
+	    "-days", "99",
+            "-in", $b_csr, "-out", $b_cert])));
+# end date before start date
+ok(run(app(["openssl", "x509", "-req", "-text",
+	      "-key", $b_key,
+	      "-not_before", "20231031000000Z",
+	      "-not_after", "today",
+              "-in", $b_csr, "-out", $b_cert])) == 1);
+
```

Comment:
Nice tests.
Would be good to explicitly check also the outcome of the three "today" settings.
It should not be too hard to get the right reference value in a portable way using some Perl function, but if so, it may be sufficient to check that all three values set in this way are equal.

Reply:
Due to my missing Perl experience I haven't found the ideal solution, yet.
I just compare if all `today` strings are equal.
I have some concerns this could fail, if the time progresses too much between two tests?

Reply:
You can use
```
use POSIX;
my $today;

$today = strftime("%Y%m%d", localtime);
...
  && $today eq substr(..., 0, 8));
```
Good point that time progressing during the test may cause hick-ups.
Yet when comparing only the dates (not hours and more fine granularity), as done above, risk should be sufficiently low.
To be even more safe, do the assignment `$today = strftime("%Y%m%d", localtime);` immediately before each test case using "today".

Reply:
IMHO the current solution is acceptable.